addon/components/rdf-input-fields/simple-value-input-field.js

```diff
@@ -14,86 +14,64 @@ export default class SimpleValueInputFieldComponent extends InputFieldComponent
     this.loadProvidedValue();
   }
 
+  get isLangStringField() {
+    return Boolean(this.args.field.language);
+  }
+
   loadProvidedValue() {
     const matches = triplesForPath(this.storeOptions);
-    const literals = matches.values.filter((value) => isLiteral(value));
-
-    if (literals.length) {
-      let literal;
 
-      if (this.args.field.language) {
-        literal = findLiteralByLanguage(literals, this.args.field.language);
-      } else {
-        // First look for the entry with no lang string.
-        // This allows us to still have multiple fields where both form:language is and is not specified
-        literal = literals.find((m) => !isLangString(m.datatype));
-
-        // For reverse compatibility: we fall back first to first matched literal value
-        // i.e. all langStrings are literals, so technically not wrong to display literal
-        //   even if it is a langString.
-        // Note: it uncovers wonky side effects,
-        //  in case we have two different literals for the same predicate.
-        //  Updating one of them might result in strange rendering behaviour.
-        // This was the case in the previous version too, and might need re-thinking. (TODO)
-        // If not clear, try it in the dummy-app
-        if (!literal) {
-          literal = literals[0];
-        }
-      }
+    if (this.isLangStringField) {
```

Comment:
This restores the old behavior if no lang string predicate is set, which should fix all the regressions we introduced.